### PR TITLE
Update API doc links

### PR DESCRIPTION
**Pull request by arthurevans**
Depends on https://github.com/PolymerElements/iron-doc-viewer/pull/184

(All of the links should basically work without it except for link to OptionalMutableData.)

### CODE REVIEW

File: app/3.0/docs/devguide/custom-css-properties.md [hunk 1]
```diff
@@ -449,7 +449,7 @@ if (ShadyCSS) {
 ```
 
 Elements using the legacy API can use the
-[`getComputedStyleValue`](/3.0/docs/api/mixins/Polymer.LegacyElementMixin#method-getComputedStyleValue)
+[`getComputedStyleValue`](/3.0/docs/api/legacy/legacy-element-mixin#LegacyElementMixin#method-getComputedStyleValue)
```
Comment (katejeffreys) [inline app/3.0/docs/devguide/custom-css-properties.md]: should be `/3.0/docs/api/legacy/legacy-element-mixin#LegacyElementMixin-method-getComputedStyleValue`?

File: app/3.0/docs/devguide/observers.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ This allows the element to avoid running observers in the default case.
 
 Like all property effects, observers are synchronous. If the observer is likely to be invoked
 frequently, consider deferring time-consuming work, like inserting or removing DOM. For example, you
-can use the [`Polymer.Async`](/{{{polymer_version_dir}}}/docs/api/namespaces/Polymer.Async) module to defer work.
+can use the [`Polymer.Async`](/{{{polymer_version_dir}}}/docs/api/utils/async) module to defer work.
```
Comment (katejeffreys) [inline app/3.0/docs/devguide/observers.md]: are utils still called `Polymer.Thing`? 
also https://github.com/Polymer/docs/issues/2557 
link itself looks fine tho  ¯\_(ツ)_/¯

File: app/3.0/docs/upgrade.md [hunk 3]
```diff
@@ -1395,7 +1395,7 @@ rather than being bundled in with `Polymer.Element`. For details, see
 
 In many cases you can use the native platform features (such as `setTimeout` or
 `requestAnimationFrame` instead of the `async` call. Polymer 2.x also provides an optional
-[`Polymer.Async`](/{{{polymer_version_dir}}}/docs/api/namespaces/Polymer.Async) module that provides
+[`Polymer.Async`](/{{{polymer_version_dir}}}/docs/api/utils/async) module that provides
```
Comment (katejeffreys) [inline app/3.0/docs/upgrade.md]: as above

File: app/3.0/docs/upgrade.md [hunk 4]
```diff
@@ -1546,7 +1546,7 @@ separate, optional imports. These include:
 -   [`<dom-bind>` element](devguide/templates#dom-bind)
 -   [`<dom-if>` element](devguide/templates#dom-if)
 -   [`<dom-repeat>` element](devguide/templates#dom-repeat)
--   [`Polymer.RenderStatus`](/{{{polymer_version_dir}}}/docs/api/namespaces/Polymer.RenderStatus)
+-   [`Polymer.RenderStatus`](/{{{polymer_version_dir}}}/docs/api/utils/render-status)
```
Comment (katejeffreys) [inline app/3.0/docs/upgrade.md]: naming of utils?

File: app/3.0/docs/devguide/custom-css-properties.md (hunk 1 above)
Comment (arthurevans) [reply app/3.0/docs/devguide/custom-css-properties.md]: Fixed.

File: app/3.0/docs/upgrade.md (hunk 4 above)
Comment (arthurevans) [reply app/3.0/docs/upgrade.md]: Oh, just realized this is the upgrade guide, which needs to be totally wiped. Will leave this one alone.

**Pull request merged**